### ISSUE TITLE
Bulk move annotations

### ISSUE THREAD

**Issue by jeremydean** [hypothesis/product-backlog#241]
### As a user I'd like to bulk move annotations from public to private to group

Especially in classrooms, students often annotate the public layer before realizing they're not in the group. These annotations cannot be moved. They have to be copied, deleted and recreated in the group layer.

I also may have made private annotations that I want to make public. While this is possible on an individual level, it would be nice to do so in bulk.

### From a user:

>After a bit of embarrassment when I realized that no one in our group could see any of the highlights I thought were shared, I attempted to go through and update the privacy settings for each one. Sadly there's no way to bulk edit them, and editing each one takes at least 5 clicks and requires changing something other than the privacy setting as well (such as adding a comment or tag). It's just not worth the trouble to do this for hundreds of highlights.

**Comment by klemay** [comment]
User request: https://hypothesis.zendesk.com/agent/tickets/2227

I've had my students using Hypothes.is for their research blogs and essays, and things are going really well. I have one student though who created his own group [redacted] instead of using that just as a tag. So now no one can see his annotations unless they joing the group  [redacted], which I don't want them to do.

Is there a way he can "transfer" his annotations from his  [redacted] group to public? His username is  [redacted]

**Comment by klemay** [comment]
User request: https://hypothesis.zendesk.com/agent/tickets/2383

>Any hope that I’ll be able to transfer my annotations to a new group so I can reuse them the next time I teach this class?

**Comment by dwhly** [comment]
> Any hope that I’ll be able to transfer my annotations to a new group so I can reuse them the next time I teach this class?

I'm not sure if others see it this way, but this could mean "Take all my annotations in this group and copy them into a new group and give me that new group ID".  A teacher might do that after they've created all the base annotations for a class, but before any students have contributed?  (i.e. preserve it like this for the next semester or additional class sections)   Feels a bit like a premium type feature?

**Comment by jeremydean** [comment]
I don't think a user would ever say "give me that new group ID."

The starter here is simply being able to move a selection of my annotations from one group I've created to another group I've created, I think.

I could see the ability to excise any replies to my previous annotations as a premium feature. But people should be able to generally move annotations from group to group in bulk.

**Comment by klemay** [comment]
>A teacher might do that after they've created all the base annotations for a class, but before any students have contributed? (i.e. preserve it like this for the next semester or additional class sections) 

In this case, the teacher wants to highlight vocabulary words in a text and provide context/definitions. He would tag those annotations with, say, #vocab. Then, when he teaches next semester, he might create a group for his new class and say "bring in all my annotations tagged #vocab from last semester".

**Comment by klemay** [comment]
Another user request today: https://hypothesis.zendesk.com/agent/tickets/2680

**Comment by klemay** [comment]
Another user request: https://hypothesis.zendesk.com/agent/tickets/2681

**Comment by klemay** [comment]
Tagging https://github.com/hypothesis/product-backlog/issues/713 as related

**Comment by klemay** [comment]
This came up again today, in reference to a private group created as part of an EdX course:
 https://hypothesis.zendesk.com/agent/filters/152272907

To sum up this ticket- 
Students have annotated in the Public layer instead of the private group, meaning that:

- Instructors can't hide annotations that are inappropriate in the class context, but which are not necessarily a violation of our community guidelines
- Students are annotating in a space which they don't perceive to be public
- It will be an inconvenience on the instructors' end to deal with asking students to migrate these annotations (and the process is a hassle to the students as well)

**Comment by clarkshahnelson** [comment]
I'm in favor! Here's my little use case: 
As a student in a doctoral program, I set up private groups for every course - just for me, to keep myself organized. At a certain point I revisit articles I have annotated in prior classes - and could use moving them to a future class/private group.

Another one: I have a private group set up for getting started on dissertation. There are some things I added in there a while ago - and some more recent. Now, working on the actual dissertation, it would be great to be able to move a set or subset of them in bulk to a new/different group so that I have them separated out from the old ones I don't need. 

At any rate, moving annotations across groups, to me, would be useful and fantastic. Else, combining groups (to eventually put together a bunch of small groups into one large one, for example, to put all the class groups from my pre-dissertation work into 1 large group) or would be super cool, too.  Thank you for all you do! :)

**Comment by clarkshahnelson** [comment]
Wait, just realized, is this what Jon's tool does here? I was looking at Facet and had forgotten there were others... :) CopyAnnotations
Copy annotations from one group to another. Also, optionally, from one domain to another.

https://jonudell.info/h/copy-annotations.mp4

https://jonudell.info/h/CopyAnnotations

https://github.com/judell/CopyAnnotations

**Comment by klemay** [comment]
@clarkshahnelson Yes, it sounds like Jon's Copy Annotations tool will meet your needs. But thank you for sharing your use case with us, as it's always helpful to hear real users' experience!
